internal/infrastructure/kubernetes/proxy_deployment.go

```diff
@@ -140,6 +140,22 @@ func expectedProxyContainers(infra *ir.Infra) ([]corev1.Container, error) {
 		},
 	}
 
+	for _, listener := range infra.Proxy.Listeners {
+		for _, lp := range listener.Ports {
+			if corev1.Protocol(lp.Protocol) == corev1.ProtocolTCP {
+				if lp.ContainerPort == envoyHTTPPort || lp.ContainerPort == envoyHTTPSPort {
+					continue
+				}
+			}
+			port := corev1.ContainerPort{
+				Name:          lp.Name,
+				ContainerPort: lp.ContainerPort,
+				Protocol:      corev1.Protocol(lp.Protocol),
+			}
+			ports = append(ports, port)
```

Comment:
can you also add some test cases in proxy_deployment_test.go ?

Reply:
done